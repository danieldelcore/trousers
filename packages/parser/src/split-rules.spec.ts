import splitRules from './split-rules';

describe('splitRules', () => {
    it('groups flat styles', () => {
        const result = splitRules(`
          .foo {
            background-color: red;
            color: blue;
          }
        `);

        expect(result).toMatchSnapshot();
    });

    it('splits multiple selectors', () => {
        const result = splitRules(`
        .foo {
          background-color: red;
          color: blue;
        }

        .foo:hover {
            background-color: green;
            color: green;
          }
        `);

        expect(result).toMatchSnapshot();
    });

    it('splits media queries selectors', () => {
        const result = splitRules(`
          .foo {
            background-color: red;
            color: blue;
          }

          @media (min-width: 768px) {
            .foo { background-color: #deecff; }
          }
        `);

        expect(result).toMatchSnapshot();
    });

    it('splits keyframes', () => {
        const result = splitRules(`
          .foo {
            background-color: red;
            color: blue;
          }

          @keyframes rotating {
              from {
                  transform: rotate(0deg);
              }
              to {
                  transform: rotate(360deg);
              }
          }
        `);

        expect(result).toMatchSnapshot();
    });

    it('empty strings should return an empty array', () => {
        {
            const result = splitRules(`

        `);

            expect(result).toEqual([]);
        }
        {
            const result = splitRules(`
          \n\t
        `);

            expect(result).toEqual([]);
        }
    });
});
