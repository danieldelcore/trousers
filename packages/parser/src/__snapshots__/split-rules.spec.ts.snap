// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitRules groups flat styles 1`] = `
Array [
  "
          .foo {
            background-color: red;
            color: blue;
          }",
]
`;

exports[`splitRules splits keyframes 1`] = `
Array [
  "
          .foo {
            background-color: red;
            color: blue;
          }",
  "

          @keyframes rotating {
              from {
                  transform: rotate(0deg);
              }
              to {
                  transform: rotate(360deg);
              }",
]
`;

exports[`splitRules splits media queries selectors 1`] = `
Array [
  "
          .foo {
            background-color: red;
            color: blue;
          }",
  "

          @media (min-width: 768px) {
            .foo { background-color: #deecff; }",
]
`;

exports[`splitRules splits multiple selectors 1`] = `
Array [
  "
        .foo {
          background-color: red;
          color: blue;
        }",
  "

        .foo:hover {
            background-color: green;
            color: green;
          }",
]
`;
